### ISSUE TITLE
Time for a single iteration of the 5B I2V model under the full-parameter SFT training setup

### ISSUE THREAD

**Issue by CacacaLalala** [a-r-r-o-w/finetrainers#65]
I am training on an H100, and a single iteration takes about 18 seconds, but in the SAT of Cogvideo-main, it only takes 8 seconds. Why is this?

**Comment by adamas-fsw** [comment]
HI! I wander ask how should I execute the code for SAT SFT, and how should the data be stored?

**Comment by CacacaLalala** [comment]
Just use finetune_single_gpu.sh or finetune_multi_gpus.sh. And the base yaml is cogvideox_i2v_5b.yaml
Here is the example:
`echo "RUN on `hostname`, CUDA_VISIBLE_DEVICES=$CUDA_VISIBLE_DEVICES"

environs="WORLD_SIZE=1 RANK=0 LOCAL_RANK=0 LOCAL_WORLD_SIZE=1"

run_cmd="$environs python train_video.py --base configs/cogvideox_5b_i2v.yaml configs/sft.yaml --seed $RANDOM"

echo ${run_cmd}
eval ${run_cmd}

echo "DONE on `hostname`"`
Data can be loaded through csv file or others.

**Comment by adamas-fsw** [comment]
Thanks for your answering, i'm confused of data structure, I wander ask how do you put the image data into dataset and how code can read it

**Comment by CacacaLalala** [comment]
I didn't use image data, only video data.

**Comment by adamas-fsw** [comment]
fine，thanks a lot！

**Comment by a-r-r-o-w** [comment]
You can remove gradnorm calculation to get a little speedup as well. I am yet to find time for profiling the scripts for where the slow synchronizations are occuring. Any insights or PRs would be appreciated!

**Comment by lijain** [comment]
> @CacacaLalala I also met the same issue, have you solved this problem. I am on A100, using deepspeed for sft, the data_worker_num is set to 4 and the training time for each iter is nearly 13s.
I have the same card with the same problem. Could you please tell me that you have solved it? Or what good method does the landlord have

**Comment by sayakpaul** [comment]
Maybe @zRzRzRzRzRzRzR could help us figure out anything obvious that we're doing wrong that is contributing to the slow-down?

**Comment by lijain** [comment]
At present, I trained on 16 cards of A100 80g, batchsize=4, picture size 384x640, and found that a row backward(loss) would take about 12s, Moreover, the processing time of vae.encode(videos).latent_dist will also take about 8s (the pre-processing of images can be tensor). At present, I do not have a good solution, may I ask you to solve?
Later, with the enlargement of the picture, the occupying time of the top two pieces will also increase greatly.

**Comment by a-r-r-o-w** [comment]
Yes, I plan to work on optimizing the scripts as soon as I find some time, but at the moment the vae encode behaviour is slow whether you start from an image or a precomputed latent (marginally a bit faster). I am unsure what causes this because for precomputed tensor, we just need to continuously stream to GPU and there is no VAE encode step involved. If you have any insights, or would like to help improve this, PRs are very welcome!

**Comment by lijain** [comment]
> backward(loss)

I really want to give opinions, as mentioned above, vae is really no way to save time. What I can't figure out is that the time of the line accelerator.backward(loss) is 12s longer with the same setting. I have located this problem before, and there is no better optimization method at present. Or do you have any ideas

**Comment by buoyancy99** [comment]
> At present, I trained on 16 cards of A100 80g, batchsize=4, picture size 384x640, and found that a row backward(loss) would take about 12s, Moreover, the processing time of vae.encode(videos).latent_dist will also take about 8s (the pre-processing of images can be tensor). At present, I do not have a good solution, may I ask you to solve? Later, with the enlargement of the picture, the occupying time of the top two pieces will also increase greatly.

Is the batch size total batch size or per GPU batch size?

**Comment by lijain** [comment]
The batch size on each gpu is 4, for a total of 16 Gpus

**Comment by buoyancy99** [comment]
> The batch size on each gpu is 4, for a total of 16 Gpus

Are you using deepspeed? I found CPU offload to slow down training significantly but it does saves memory to make batch size 4 possible

**Comment by lijain** [comment]
Have you tried what iter is without the cpu? Some time ago, I found that the acceleration may not be used in the calculation of attn, and I will change it later when there is time
